src/pages/register/RegisterPage.tsx

```diff
@@ -0,0 +1,104 @@
+import { useMutation } from "@tanstack/react-query";
+import { FormProvider, useForm } from "react-hook-form";
+import { register } from "../../api/auth/users";
+import { useNavigate } from "react-router-dom";
+import { LABEL, TEXT } from "../../utils/strings";
+import { ROUTES } from "../../Routes/routes";
+import InputField, {
+  INPUT_FIELD_VARIANTS,
+} from "../../components/inputs/InputField";
+import AuthForm from "../../components/authForm/AuthForm";
+import { RegisterData } from "../../types/auth";
+import { InputTypeEnum } from "../../components/inputs/inputFieldUtils";
+import { TEST_ID } from "../../components/inputs/__tests__/testIds";
+import { dataName } from "./types/types";
+import { UseToastContext } from "../../context/toastContext/ToastContext";
+import { ToastSeverity } from "../../components/toast/Toast";
+import { useState } from "react";
+
+type Errors = {
+  email?: string;
+  password?: string;
+};
+
+type RegisterBadRequest = {
+  error: string;
+  errors: Errors[];
+  statusCode: string;
+  message?: string;
+};
+
+export default function RegisterPage() {
+  const [error, setError] = useState<Errors>({});
+  const navigate = useNavigate();
+  const registerMethods = useForm<RegisterData>();
+  const { toastHandler } = UseToastContext();
+
+  const { mutate, isPending } = useMutation({
+    mutationFn: async (data: RegisterData) => await register(data),
+    onSuccess: (_success) => {
+      toastHandler({
+        message: _success.message,
+        severity: ToastSeverity.SUCCESS,
+      });
+      navigate(ROUTES.LOGIN);
+    },
+    onError: async (_fail: RegisterBadRequest) => {
+      toastHandler({
+        message: _fail.message || _fail.statusCode, //need to work on backend to format the erros
```

Comment:
Usually there's not a good practice to leave comments in the code unless it's something that might be forgotten in time (specific logic)
You can create a small ticket for that

But if the comment is needed, you can add the **todo** keyword
```suggestion
        message: _fail.message || _fail.statusCode, //todo: need to work on backend to format the errors
```

Reply:
forgot to remove it, i implemented the fix in the backend to also display "email already in use" under the email input. Will fix in the next commit